File: app/src/main/java/net/harimurti/tv/data/License.java
package net.harimurti.tv.data;

import android.os.Parcel;
import android.os.Parcelable;

public class License implements Parcelable {
    public String domain;
    public String drm_url;

    protected License(Parcel in) {
        domain = in.readString();
        drm_url = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(domain);
        dest.writeString(drm_url);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<License> CREATOR = new Creator<License>() {
        @Override
        public License createFromParcel(Parcel in) {
            return new License(in);
        }

        @Override
        public License[] newArray(int size) {
            return new License[size];
        }
    };
}

File: app/src/main/java/net/harimurti/tv/model/License.kt
package net.harimurti.tv.model

import android.os.Parcel
import android.os.Parcelable

open class License protected constructor(`in`: Parcel) : Parcelable {
    var domain: String = `in`.readString().toString()
    var drm_url: String = `in`.readString().toString()

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(domain)
        dest.writeString(drm_url)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<License?> = object : Parcelable.Creator<License?> {
            override fun createFromParcel(`in`: Parcel): License {
                return License(`in`)
            }

            override fun newArray(size: Int): Array<License?> {
                return arrayOfNulls(size)
            }
        }
    }

}